File: admin/src/api/product.ts
import request from '@/utils/request'
import type { ApiResponse, PageResult } from '@/utils/types'

export interface Product {
  id: number
  category_id: number
  name: string
  subtitle?: string
  cover: string
  images: string[]
  description?: string
  price: number
  original_price?: number
  stock: number
  sales: number
  status: number
  is_hot: number
  is_new: number
  sort: number
  created_at: string
}

export interface Category {
  id: number
  parent_id: number
  name: string
  icon?: string
  sort: number
  status: number
  children?: Category[]
}

export interface ProductParams {
  page?: number
  size?: number
  category_id?: number
  status?: number
  keyword?: string
}

export const getProductList = (params: ProductParams) => {
  return request.get<ApiResponse<PageResult<Product>>>('/admin/products', { params })
}

export const getProductDetail = (id: number) => {
  return request.get<ApiResponse<Product>>(`/admin/products/${id}`)
}

export const createProduct = (data: Partial<Product>) => {
  return request.post<ApiResponse<Product>>('/admin/products', data)
}

export const updateProduct = (id: number, data: Partial<Product>) => {
  return request.put<ApiResponse<Product>>(`/admin/products/${id}`, data)
}

export const deleteProduct = (id: number) => {
  return request.delete<ApiResponse>(`/admin/products/${id}`)
}

export const getCategoryList = () => {
  return request.get<ApiResponse<Category[]>>('/admin/categories')
}

export const createCategory = (data: Partial<Category>) => {
  return request.post<ApiResponse<Category>>('/admin/categories', data)
}

export const updateCategory = (id: number, data: Partial<Category>) => {
  return request.put<ApiResponse<Category>>(`/admin/categories/${id}`, data)
}

export const deleteCategory = (id: number) => {
  return request.delete<ApiResponse>(`/admin/categories/${id}`)
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional, List


class ProductCreate(BaseModel):
    category_id: int
    name: str
    subtitle: Optional[str] = None
    cover: Optional[str] = None
    images: List[str] = []
    description: Optional[str] = None
    price: float
    original_price: Optional[float] = None
    stock: int = 0
    is_hot: bool = False
    is_new: bool = False
    status: int = 1
    sort: int = 0


class ProductUpdate(BaseModel):
    category_id: Optional[int] = None
    name: Optional[str] = None
    subtitle: Optional[str] = None
    cover: Optional[str] = None
    images: Optional[List[str]] = None
    description: Optional[str] = None
    price: Optional[float] = None
    original_price: Optional[float] = None
    stock: Optional[int] = None
    is_hot: Optional[bool] = None
    is_new: Optional[bool] = None
    status: Optional[int] = None
    sort: Optional[int] = None


class CategoryCreate(BaseModel):
    parent_id: int = 0
    name: str
    icon: Optional[str] = None
    sort: int = 0
    status: int = 1


class CategoryUpdate(BaseModel):
    parent_id: Optional[int] = None
    name: Optional[str] = None
    icon: Optional[str] = None
    sort: Optional[int] = None
    status: Optional[int] = None
